src/Addon/PluginUpgrade.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace GiveAddon\Addon\Helpers;
+
+/**
+ * Class PluginUpgrade
+ * @package GiveAddon\Infrastructure
+ *
+ * @unreleased
+ */
+class PluginUpgrade {
+    /**
+     * @unreleased
+     */
+    public function storePluginUpgradeVersion() {
+        $pluginVersion = preg_replace( '/[^0-9.].*/', '', get_option( 'give_addon_version' ) );
+
+        // Is Fresh install?
+        if ( ! $pluginVersion ) {
+            $pluginVersion = '1.0.0';
+        }
+
+        if ( version_compare( $pluginVersion, ADDON_CONSTANT_VERSION, '<' ) ) {
```

Comment:
I would suggest abstracting this check for the purpose of readability. I find that it is easy to forget how this comparison work, given the order of the arguments. Encapsulating this logic into a function will provide a function name that explains the intended functionality.